src/ProjectService.Business/Commands/Task/CreateTaskPropertyCommand.cs

```diff
@@ -0,0 +1,66 @@
+﻿using LT.DigitalOffice.Kernel.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Enums;
+using LT.DigitalOffice.Kernel.Exceptions.Models;
+using LT.DigitalOffice.Kernel.Extensions;
+using LT.DigitalOffice.Kernel.FluentValidationExtensions;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.ProjectService.Business.Commands.Interfaces;
+using LT.DigitalOffice.ProjectService.Data.Interfaces;
+using LT.DigitalOffice.ProjectService.Mappers.RequestsMappers.Interfaces;
+using LT.DigitalOffice.ProjectService.Models.Dto.Requests;
+using LT.DigitalOffice.ProjectService.Validation.Interfaces;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace LT.DigitalOffice.ProjectService.Business.Commands
+{
+    public class CreateTaskPropertyCommand : ICreateTaskPropertyCommand
+    {
+        private readonly IDbTaskPropertyMapper _mapper;
+        private readonly IAccessValidator _accessValidator;
+        private readonly IUserRepository _userRepository;
+        private readonly ICreateTaskPropertyValidator _validator;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly ITaskPropertyRepository _taskProperyRepository;
+
+        public CreateTaskPropertyCommand(
+            IDbTaskPropertyMapper mapper,
+            IAccessValidator accessValidator,
+            ITaskPropertyRepository taskProperyRepository,
+            IUserRepository userRepository,
+            ICreateTaskPropertyValidator validator,
+            IHttpContextAccessor httpContextAccessor)
+        {
+            _mapper = mapper;
+            _validator = validator;
+            _userRepository = userRepository;
+            _accessValidator = accessValidator;
+            _taskProperyRepository = taskProperyRepository;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public OperationResultResponse<IEnumerable<Guid>> Execute(CreateTaskPropertyRequest request)
+        {
+            var userId = _httpContextAccessor.HttpContext.GetUserId();
+
+            if (!(_accessValidator.IsAdmin() || _userRepository.AreUserProjectExist(userId, request.ProjectId)))
```

Comment:
not all users in project has rights for creating properties for tasks

Reply:
it will need rewrite in soon feature

Reply:
say @MariyaSin to create task for it